themes/origins/assets/component-cart-drawer-items.js

```diff
@@ -1,10 +1,139 @@
 class CartDrawer extends HTMLElement {
+  constructor() {
+    super();
+
+    this.cart = document.querySelector("yc-drawer#cart");
+    this.cartBubble = this.cart.querySelector("[data-cart-badge]");
+    this.subTotals = this.cart.querySelectorAll("[data-total]");
+  }
+
   connectedCallback() {
     this._render();
   }
 
   _render() {
-    return 0;
+    subscribe(PUB_SUB_EVENTS.cartUpdate, (payload) => {
+      const { count, sub_total, items } = payload.cartData;
+
+      if (payload.source === "product-form") this.cart.open();
+
+      this.updateCartBubble(count, sub_total);
+      this.updateCartList(items);
+    });
+  }
+
+  updateCartSubTotal(subtotal) {
+    this.subTotals.forEach((node) => {
+      node.textContent = formatCurrency(subtotal);
+    });
+  }
+
+  updateCartCount(count) {
+    if (this.cartBubble) {
+      this.cartBubble.textContent = count;
+      this.cartBubble.classList.toggle("hidden", count === 0);
+    }
+  }
+
+  updateCartBubble(count, subtotal) {
+    this.updateCartSubTotal(subtotal);
+    this.updateCartCount(count);
+  }
+
+  updateCartList(items) {
+    const fragment = new DocumentFragment();
+    const template = this.cart.querySelector("#cart-item-template");
+
+    items.map((item) => {
+      const cartItem = this.createCartItem(template, item);
+      fragment.append(cartItem);
+    });
+
+    this.updateDrawerState();
+    this.replaceContent(fragment);
+  }
+
+  createCartItem(template, item) {
+    const cartItem = template.content.cloneNode(true);
+    const elements = this.getCartItemElements(cartItem);
+
+    this.updateItemImage(elements.image, item.productVariant.product);
+    this.updateItemTitle(elements.title, item.productVariant.product);
+    this.updateItemVariant(elements.variant, item.productVariant.variations);
+    this.updateItemQuantity(elements.quantity, item.quantity);
+    this.updateItemPrice(elements.price, item.price);
+
+    return cartItem;
+  }
+
+  getCartItemElements(cartItem) {
+    const [image, title, variant, price, quantity] = cartItem.querySelectorAll(
+      "[data-cart-item-image], [data-cart-item-title], [data-cart-item-variant], [data-price], [data-quantity]",
+    );
```

Comment:
:lets-talk:

```suggestion
    const [image, title, variant, price, quantity] = cartItem.querySelectorAll("[data-cart-item]");
```

**EX:**
```liquid
<div class="image" data-cart-item="image">
...
</div>
```